handleleave.go

```diff
@@ -106,12 +106,11 @@ type CurrentStateQuerier interface {
 	CurrentStateEvent(ctx context.Context, roomID spec.RoomID, eventType string, stateKey string) (PDU, error)
 }
 
-// HandleSendLeave handles requests to `/send_leave
-// Returns the parsed event and an error.
-func HandleSendLeave(ctx context.Context,
-	requestContent []byte,
+// handleSendLeave handles requests to `/send_leave
+// Returns the parsed event or an error.
+func handleSendLeave(ctx context.Context,
+	event PDU,
```

Comment:
The proposal I mentioned in https://github.com/matrix-org/gomatrixserverlib/pull/387#discussion_r1214123648 was to keep this function public, and to serve as the entry point for callers, because:

> Whilst the caller could call roomVer.HandleSendLeave this then creates bad symmetry, as there are cases where you don't know the room version at this point (e.g invites).